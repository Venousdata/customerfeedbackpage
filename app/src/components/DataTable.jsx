// this component just displays the reviews in a table format,
// allowing for sorting on columns, filtering, etc.

import { makeStyles } from "@material-ui/core";
import { DataGrid } from "@material-ui/data-grid";
import { getColumns, getRows } from "../helpers/helpers";

const useStyles = makeStyles((theme) => ({
  tableContainer: { padding: "1rem", width: "100%", height: "350px" },
}));

const DataTable = ({ reviews }) => {
  const columns = getColumns(reviews, ["id", "productid"]);
  const rows = getRows(reviews);
  const classes = useStyles();

  return (
    <div className={classes.tableContainer}>
      <DataGrid
        columns={columns}
        rows={rows}
        pageSize={5}
        disableSelectionOnClick
      />
    </div>
  );
};

export default DataTable;
