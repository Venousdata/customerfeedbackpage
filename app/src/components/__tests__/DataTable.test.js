import React from "react";
import { render } from "@testing-library/react";
import DataTable from "../DataTable";

describe("DataTable tests", () => {
  it("DataTable should render and match snapshot", (done) => {
    const container = render(<DataTable />);
    expect(container.firstChild).toMatchSnapshot();
    done();
  });
});
