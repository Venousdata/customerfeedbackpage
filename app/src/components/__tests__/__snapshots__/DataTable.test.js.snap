// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable tests DataTable should render and match snapshot 1`] = `undefined`;
